src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -186,134 +182,79 @@ function geometryToLine(geometry, properties, options, multiGeomAttributes) {
     }
 }
 
-function geometryToPolygon(geometry, properties, options, multiGeomAttributes) {
-    const vertices = multiGeomAttributes ? multiGeomAttributes.vertices : geometry.vertices;
-
+function geometryToPolygon(feature, properties, options) {
     // get altitude / color from properties
-    const altitude = getAltitude(options, properties, vertices);
+    const altitude = getAltitude(options, properties, feature.vertices);
     const color = getColor(options, properties);
 
     const geom = prepareBufferGeometry(
-        vertices,
+        feature.vertices,
         color,
         altitude);
 
-    const indices = [];
-    // Build indices
-    if (multiGeomAttributes) {
-        // Multi polygon case
-        for (let i = 0; i < geometry.length; i++) {
-            const holesOffsets = geometry[i].holes.map(h => h.offset);
-            const start = multiGeomAttributes.elements[i].offset + geometry[i].contour.offset;
-            const end = multiGeomAttributes.elements[i].offset + multiGeomAttributes.elements[i].count;
-            const triangles = Earcut(geom.attributes.position.array.slice(start * 3, end * 3),
-                holesOffsets,
-                3);
-            for (const indice of triangles) {
-                indices.push(start + indice);
-            }
-        }
-    } else {
-        // Single polygon case
-        const holesOffsets = geometry.holes.map(h => h.offset);
-        const triangles = Earcut(geom.attributes.position.array, holesOffsets, 3);
-        for (const indice of triangles) {
-            indices.push(indice);
-        }
+    let indices = [];
+    for (const geometry of feature.geometry) {
+        const start = geometry.indices[0];
+        const end = geometry.indices.slice(-2).reduce((a, b) => (a + b));
+
+        // holes are starting at index 2, and offset are the first element
+        const holesOffsets = geometry.indices.slice(2)
+            .filter((e, i) => (i % 2 == 0))
+            .map(i => i - start);
+
+        const triangles = Earcut(geom.attributes.position.array.slice(start * 3, end * 3),
+                holesOffsets, 3);
+
+        indices = indices.concat(triangles.map(i => i + start));
     }
 
     geom.setIndex(new THREE.BufferAttribute(new Uint16Array(indices), 1));
     return new THREE.Mesh(geom);
 }
 
 
-function geometryToExtrudedPolygon(geometry, properties, options, multiGeomAttributes) {
-    const vertices = multiGeomAttributes ? multiGeomAttributes.vertices : geometry.vertices;
-
+function geometryToExtrudedPolygon(feature, properties, options) {
```

Comment:
You could rename this method since it now takes a `feature` instead of a `geometry`

Reply:
Done